### PR TITLE
Replaced bit.ly links and updated elasticsearch.org to elastic.co.

### PR DESCRIPTION
**Pull request by debadair**
First part of the phase 1 cleanup of the Definitive Guide per https://github.com/elastic/x-plugins/issues/986.

### CODE REVIEW

File: 010_Intro/10_Installing_ES.asciidoc [hunk 1]
```diff
@@ -8,22 +8,22 @@ Preferably, you should install the latest version of the((("Java", "installing")
 from http://www.java.com[_www.java.com_].
 
 You can download the latest version of Elasticsearch from
-http://www.elasticsearch.org/download/[_elasticsearch.org/download_].
+https://www.elastic.co/downloads/elasticsearch[_elasticsearch.co/downloads/elasticsearch_].
```
Comment (clintongormley) [inline 010_Intro/10_Installing_ES.asciidoc]: elasticsearch -> elastic

File: 010_Intro/15_API.asciidoc [hunk 2]
```diff
@@ -30,7 +30,7 @@ otherwise, they may not be able to understand each other.
 ====
 
 More information about the Java clients can be found in the Java API section
-of the http://www.elasticsearch.org/guide/[Guide].
+of the http://www.elastic.co/guide/[Elastic documentation].
```
Comment (clintongormley) [inline 010_Intro/15_API.asciidoc]: It's now in the clients section, perhaps link to https://www.elastic.co/guide/en/elasticsearch/client/index.html instead?

File: 010_Intro/15_API.asciidoc [hunk 3]
```diff
@@ -42,7 +42,7 @@ seen, you can even talk to Elasticsearch from the command line by using the
 NOTE: Elasticsearch provides official clients((("clients", "other than Java"))) for several languages--Groovy,
 JavaScript, .NET, PHP, Perl, Python, and Ruby--and there are numerous
 community-provided clients and integrations, all of which can be found in the
-http://www.elasticsearch.org/guide/[Guide].
+http://www.elastic.co/guide/[Elastic documentation].
```
Comment (clintongormley) [inline 010_Intro/15_API.asciidoc]: Link to https://www.elastic.co/guide/en/elasticsearch/client/index.html instead?

File: 010_Intro/20_Document.asciidoc [hunk 4]
```diff
@@ -51,7 +51,7 @@ a flat table structure.
 ====
 Almost all languages have modules that will convert arbitrary  data
 structures or objects((("JSON", "converting your data to"))) into JSON for you, but the details are specific  to each
-language. Look for modules that handle JSON _serialization_ or _marshalling_. http://www.elasticsearch.org/guide[The official
+language. Look for modules that handle JSON _serialization_ or _marshalling_. http://www.elastic.co/guide[The official
```
Comment (clintongormley) [inline 010_Intro/20_Document.asciidoc]: Clients link https://www.elastic.co/guide/en/elasticsearch/client/index.html

File: 052_Mapping_Analysis/40_Analysis.asciidoc [hunk 5]
```diff
@@ -202,7 +202,7 @@ that the original word occupied in the original string.
 
 TIP: The `type` values like `<ALPHANUM>` vary ((("types", "type values returned by analyzers")))per analyzer and can be ignored.
 The only place that they are used in Elasticsearch is in the
-http://www.elasticsearch.org/guide/en/elasticsearch/guide/current/analysis-intro.html#analyze-api[`keep_types` token filter].
+http://www.elastic.co/guide/en/elasticsearch/guide/current/analysis-intro.html#analyze-api[`keep_types` token filter].
```
Comment (clintongormley) [inline 052_Mapping_Analysis/40_Analysis.asciidoc]: {ref}/analysis-keep-types-tokenfilter.html

File: 070_Index_Mgmt/10_Settings.asciidoc [hunk 6]
```diff
@@ -2,7 +2,7 @@
 
 There are many many knobs((("index settings"))) that you can twiddle to
 customize index behavior, which you can read about in the
-http://www.elasticsearch.org/guide/en/elasticsearch/guide/current/_index_settings.html#_index_settings[Index Modules reference documentation],
+http://www.elastic.co/guide/en/elasticsearch/guide/current/_index_settings.html#_index_settings[Index Modules reference documentation],
```
Comment (clintongormley) [inline 070_Index_Mgmt/10_Settings.asciidoc]: {ref}/index-modules.html

File: 070_Index_Mgmt/40_Custom_Dynamic_Mapping.asciidoc [hunk 7]
```diff
@@ -60,7 +60,7 @@ a `date` field, you have to add it manually.
 [NOTE]
 ====
 Elasticsearch's idea of which strings look like dates can be altered
-with the http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/mapping-root-object-type.html[`dynamic_date_formats` setting].
+with the {ref}/mapping-root-object-type.html[`dynamic_date_formats` setting].
```
Comment (clintongormley) [inline 070_Index_Mgmt/40_Custom_Dynamic_Mapping.asciidoc]: {ref}/dynamic-field-mapping.html#date-detection

File: 070_Index_Mgmt/40_Custom_Dynamic_Mapping.asciidoc [hunk 8]
```diff
@@ -140,4 +140,4 @@ The `unmatch` and `path_unmatch` patterns((("unmatch pattern")))((("path_unmap p
 that would otherwise match.
 
 More configuration options can be found in the
-http://bit.ly/1wdHOzG[reference documentation for the root object].
+{ref}/root-object.html#_properties[reference documentation for the root object].
```
Comment (clintongormley) [inline 070_Index_Mgmt/40_Custom_Dynamic_Mapping.asciidoc]: {ref}/dynamic-mapping.html[dynamic mapping documentation].

File: 080_Structured_Search/40_bitsets.asciidoc [hunk 9]
```diff
@@ -78,7 +78,7 @@ doesn't make sense to do so:
 
 Script filters::
 
-The results((("script filters, no caching of results"))) from http://www.elasticsearch.org/guide/en/elasticsearch/guide/current/filter-caching.html#_controlling_caching[`script` filters] cannot
+The results((("script filters, no caching of results"))) from http://www.elastic.co/guide/en/elasticsearch/guide/current/filter-caching.html#_controlling_caching[`script` filters] cannot
```
Comment (clintongormley) [inline 080_Structured_Search/40_bitsets.asciidoc]: {ref}/query-dsl-script-filter.html

File: 300_Aggregations/75_sigterms.asciidoc [hunk 10]
```diff
@@ -16,7 +16,7 @@ PUT /_snapshot/sigterms <1>
 {
     "type": "url",
     "settings": {
-        "url": "http://download.elasticsearch.org/definitiveguide/sigterms_demo/"
+        "url": "http://download.elastic.co/definitiveguide/sigterms_demo/"
```
Comment (clintongormley) [inline 300_Aggregations/75_sigterms.asciidoc]: @polyfractal is this URL correct?

File: 080_Structured_Search/40_bitsets.asciidoc (hunk 9 above)
Comment (clintongormley) [inline 080_Structured_Search/40_bitsets.asciidoc]: Or this?

    {ref}//query-dsl-script-query.html

**Comment by clintongormley** [comment]
Thanks for doing this @debadair!

I haven't tested the links, but at least now they should fail the build process if they're not correct

File: 010_Intro/15_API.asciidoc (hunk 2 above)
Comment (debadair) [inline 010_Intro/15_API.asciidoc]: Done

File: 010_Intro/15_API.asciidoc (hunk 3 above)
Comment (debadair) [inline 010_Intro/15_API.asciidoc]: Done

File: 010_Intro/20_Document.asciidoc (hunk 4 above)
Comment (debadair) [inline 010_Intro/20_Document.asciidoc]: Done

File: 052_Mapping_Analysis/40_Analysis.asciidoc (hunk 5 above)
Comment (debadair) [inline 052_Mapping_Analysis/40_Analysis.asciidoc]: Done

File: 070_Index_Mgmt/10_Settings.asciidoc (hunk 6 above)
Comment (debadair) [inline 070_Index_Mgmt/10_Settings.asciidoc]: Done

File: 070_Index_Mgmt/40_Custom_Dynamic_Mapping.asciidoc (hunk 7 above)
Comment (debadair) [inline 070_Index_Mgmt/40_Custom_Dynamic_Mapping.asciidoc]: Done

File: 070_Index_Mgmt/40_Custom_Dynamic_Mapping.asciidoc (hunk 8 above)
Comment (debadair) [inline 070_Index_Mgmt/40_Custom_Dynamic_Mapping.asciidoc]: Done

File: 080_Structured_Search/40_bitsets.asciidoc (hunk 9 above)
Comment (debadair) [inline 080_Structured_Search/40_bitsets.asciidoc]: Used the latter.

File: 300_Aggregations/75_sigterms.asciidoc (hunk 10 above)
Comment (debadair) [inline 300_Aggregations/75_sigterms.asciidoc]: This URL does not work. Where is this dataset, really?

Comment (debadair) [inline 300_Aggregations/75_sigterms.asciidoc]: Pinged @drewr about this. Going to assume that we'll get it fixed on the site side.

**Comment by debadair** [comment]
Thanks @clintongormley. I've updated the links per your comments & I think we're good to go with this set of changes. Will monitor the situation with the sample dataset & watch for any build breakages due to flaky cross-doc links.

File: 300_Aggregations/75_sigterms.asciidoc (hunk 10 above)
Comment (debadair) [inline 300_Aggregations/75_sigterms.asciidoc]: The URL is in fact correct and live.

**Pull request merged**